src/url.rs

```diff
@@ -12,334 +12,45 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-use crate::errors::{Error, Result};
-use serde::{Deserialize, Deserializer, Serialize, Serializer};
-use std::fmt;
+/// FIXME - remove and complete refactoring after connector and troy branches are merged
 
-/// The type or resource the url references
-#[derive(Clone, Copy, Eq, PartialEq, Hash, Debug)]
-pub enum ResourceType {
-    /// This is a pipeline
-    Pipeline,
-    /// This is an onramp
-    Onramp,
-    /// This is an offramp
-    Offramp,
-    /// This is a binding
-    Binding,
-}
+/// Deprecated - Alias for `tremor_common::url::ResourceType`
+pub type ResourceType = tremor_common::url::ResourceType;
 
-/// The scrope of the URL
-#[derive(Clone, Copy, Eq, PartialEq, Hash, Debug)]
-pub enum Scope {
-    /// This URL identifies a specific port
-    Port,
-    /// This URL identifies a servant (a running instance)
-    Servant,
-    /// This URL identifes an artefact (a non running configuration)
-    Artefact,
-    /// This URL identifies a type of artefact
-    Type,
-}
+/// Deprecated - Alias for `tremor_common::url::Scope`
+pub type Scope = tremor_common::url::Scope;
 
 /// module for standard port names
 pub mod ports {
     use beef::Cow;
 
     /// standard input port
-    pub const IN: Cow<'static, str> = Cow::const_str("in");
+    pub const IN: Cow<'static, str> = tremor_common::url::ports::IN;
 
     /// standard output port
-    pub const OUT: Cow<'static, str> = Cow::const_str("out");
+    pub const OUT: Cow<'static, str> = tremor_common::url::ports::OUT;
 
     /// standard err port
-    pub const ERR: Cow<'static, str> = Cow::const_str("err");
+    pub const ERR: Cow<'static, str> = tremor_common::url::ports::ERR;
 
     /// standard metrics port
-    pub const METRICS: Cow<'static, str> = Cow::const_str("metrics");
+    pub const METRICS: Cow<'static, str> = tremor_common::url::ports::METRICS;
```

Comment:
Could be easier with:
```
pub use tremor_common::url::ports::*;
```

Reply:
Used the first as this wrapper/redirect will disappear with the connectors rebase.